examples/opt_serving/interactive_hosted.py

```diff
@@ -49,15 +56,26 @@ def batching_loop(timeout=TIMEOUT_MS, max_tokens=MAX_BATCH_TOKENS, max_bs=MAX_BS
     # TODO(roller):
     # - group by generation type, topp etc, as we cannot share these
     # - modify timeout logic to be cumulative
-    global BATCH_QUEUE
+    global GENERATE_BATCH_QUEUE
+    global LOGPROBS_BATCH_QUEUE
+    global PAST_CACHE
+    global PAST_CACHE_TIMEOUT
+    global PAST_CACHE_SIZE_LIMIT
 
-    batch = []
+    generate_batch = []
+    logprobs_batch = []
     while True:
-        try:
-            # for now, we only have 1 worker, so can always index to shard 0
-            target_queue = BATCH_QUEUE.queue_shards[0].get_largest_queue()
-            if not target_queue:
-                continue
+        generate_batch, made_model_call = generate_loop(generate_batch, timeout=timeout, max_tokens=max_tokens, max_bs=max_bs)
```

Comment:
I suppose this means if the external users are submitting requests, the returned `made_model_call` of their requests will be True. So your endpoint won't run?

But if at any point, you are submitting requests while any other users are not submitting requests (hence the old batching queue is empty), your requests will go through into the `if not made_model_call` branch, which will then go to the function you added in `generator.py`? 

Please correct if some of my understanding is something wrong.

Reply:
Yeah, Zhuohan suggested that I prioritize the existing endpoint, so I implemented this logic to prioritize requests to completions and only run my endpoint if the queue is empty.